### ISSUE TITLE
Use TCP frontend and HTTP backend

### ISSUE THREAD

**Issue by mielmat** [jcmoraisjr/haproxy-ingress#930]
Hello,
I need some help with configuring haproxy ingress. I used version 0.7 of controller, and  that one created TCP frontend bound to port 443, then TCP backend which had 'server' directive with unix socket and then another pair of frontend - backend in HTTP mode. Then I upgraded to latest version and things stopped working, as by default all frontend were created in HTTP mode. I used 
```yaml
ingress.kubernetes.io/tcp-service-port: "443"
    ingress.kubernetes.io/config-tcp-service: |
    ....
```
in my k8s ingress definition, but https frontend is also bounded to port 443 - so now traffic is load balanced by both frontends - tcp and http. Is there any way to make the configuration similar to the old one? Any proxying option can be used to redirect all traffic to tcp frontend on port 443 and then do the routing to proper backends?

Thanks in advance

**Comment by jessequinn** [comment]
Refer to https://haproxy-ingress.github.io/docs/configuration/keys/#bind i believe.

change the bind ports

**Comment by jcmoraisjr** [comment]
Up to v0.7 the controller added a static proxy configuration, which generates a relevant latency. Since v0.8 this tcp proxy is only added when needed, e.g. when you have at least one ssl-passthrough configured.

I didn't understand why you needed that proxy. What configuration are you trying to achieve?

If you just need to pass ssl though the proxy, you can use ssl-passthrough which is pretty much the same as configure tcp-service-port in :443 and use tls over it. If it's not the case you can change the bind port as stated by @jessequinn , currently the controller doesn't allow you to turn http or https ports off.

**Comment by mielmat** [comment]
ok, so I dig a bit deeper into that, and I'm considering something like this:
I can change https port binding to something different (for example port 8443), and then stick with listening TCP requests on port 443. Part of the system which is working with TCP works in that case, but I have other services exposed which works with HTTP. So, I would like to forward requests (based on SNI) to the frontend which can consume HTTP requests. How can I achieve that? I tried to add to my tcp backend:
```yaml
ingress.kubernetes.io/config-backend: |                                                                                                                                                             
  server default_https 127.0.0.1:8443
```
but that is not added to haproxy.cfg
Case is similar to this one: https://www.liip.ch/en/blog/haproxy-selective-tls-termination

**Comment by jcmoraisjr** [comment]
It seems to me that you'll have the configuration you're expecting if you configure ssl-passthrough. When this configuration is found by the controller, it'll create that same proxy you could find on older versions. Give it a try, make a poc, see how the controller behaves. Requests to :443 will be routed based on SNI, which is the only way to share a single port between more than one service without ssl-offloading the request.

**Comment by mielmat** [comment]
Hello, I want to summarize the steps which I performed to achieve expected results:
so yes, thanks for mentioning ssl-passthrough option, as that did the trick partially, as I had TCP listener in front of everything. I had to add dummy ingress to add the annotation (as TLS for other needs to be handled by HaProxy; or maybe there is another way to do that?). After doing that, it didn't work 100% according to requirements, so I had to dig deeper. At the end, what helped me a lot was using custom-section parameter, to make additional layer similar to old implementation of the controller (I added another frontend bound to unix socket and backend which listen to it). As my use case was quite strange, and I had to get a lot of info from different people, this way was the easiest thing to do (it is not perfect as it will be hard to maintain, but at the end it seems to work). Of course entire round of testing is ahead of me, but for now it is quite promising.
